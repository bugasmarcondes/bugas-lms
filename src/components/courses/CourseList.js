import React from "react";

const CourseList = ({ courses, handleDelete }) => (
  <table className="striped responsive-table centered">
    <thead>
      <tr>
        <th>Name</th>
        <th></th>
      </tr>
    </thead>
    <tbody>
      {courses && courses.length > 0 ? (
        courses.map((c) => (
          <tr key={c.id}>
            <td>
              <a href={"/course/" + c.id}>{c.title}</a>
            </td>
            <td>
              <button
                className="btn waves-effect waves-light"
                onClick={() => handleDelete(c.id)}
              >
                Remove
                <i className="material-icons right">delete</i>
              </button>
            </td>
          </tr>
        ))
      ) : (
        <tr>
          <td>No courses yet!</td>
        </tr>
      )}
    </tbody>
  </table>
);

export default CourseList;
