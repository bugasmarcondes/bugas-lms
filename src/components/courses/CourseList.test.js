import React from "react";
import CourseList from "./CourseList";
import renderer from "react-test-renderer";

const courses = [];

it("check if message is displayed when no courses are listed yet", () => {
  const tree = renderer.create(<CourseList courses={courses} />);

  expect(tree).toMatchSnapshot();
});
