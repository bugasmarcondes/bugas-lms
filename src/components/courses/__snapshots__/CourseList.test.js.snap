// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if message is displayed when no courses are listed yet 1`] = `
<table
  className="striped responsive-table centered"
>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th />
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        No courses yet!
      </td>
    </tr>
  </tbody>
</table>
`;
